graphql/resolvers.go

```diff
@@ -441,6 +444,131 @@ func (r *queryResolver) TaskFiles(ctx context.Context, taskID string) ([]*Groupe
+	}
+
+	apiEventLogPointers := []*restModel.APIEventLogEntry{}
+	for _, e := range filteredEvents {
+		apiEventLog := restModel.APIEventLogEntry{}
+		err = apiEventLog.BuildFromService(&e)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Unable to build APIEventLogEntry from EventLog: %s", err.Error()))
+		}
+		apiEventLogPointers = append(apiEventLogPointers, &apiEventLog)
+	}
+
+	t, err := r.sc.FindTaskById(taskID)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("error finding task by id %s: %s", taskID, err.Error()))
+	}
+	if t == nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("cannot find task with id %s", taskID))
+	}
+	p, err := r.sc.FindProjectById(t.Project)
+	if p == nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("could not find project '%s'", t.Project))
+	}
+
+	defaultLogger := p.DefaultLogger
+	if defaultLogger == "" {
+		defaultLogger = evergreen.GetEnvironment().Settings().LoggerConfig.DefaultLogger
+	}
+
+	taskLogPointers := []*apimodels.LogMessage{}
+	systemLogPointers := []*apimodels.LogMessage{}
+	agentLogPointers := []*apimodels.LogMessage{}
+
+	if defaultLogger == model.BuildloggerLogSender {
+		// get logs from cedar
+		// task logs
+		taskLogReader, err := apimodels.GetBuildloggerLogs(ctx, evergreen.GetEnvironment().Settings().LoggerConfig.BuildloggerBaseURL, taskID, apimodels.TaskLogPrefix, LogMessageCount, t.Execution)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, err.Error())
+		}
+		taskLogs := apimodels.ReadBuildloggerToSlice(ctx, taskID, taskLogReader)
+		for i := range taskLogs {
+			taskLogPointers = append(taskLogPointers, &taskLogs[i])
+		}
+
+		// system logs
+		systemLogReader, err := apimodels.GetBuildloggerLogs(ctx, evergreen.GetEnvironment().Settings().LoggerConfig.BuildloggerBaseURL, taskID, apimodels.SystemLogPrefix, LogMessageCount, t.Execution)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, err.Error())
+		}
+		systemLogs := apimodels.ReadBuildloggerToSlice(ctx, taskID, systemLogReader)
+		for i := range systemLogs {
+			systemLogPointers = append(systemLogPointers, &systemLogs[i])
+		}
+
+		// agent logs
+		agentLogReader, err := apimodels.GetBuildloggerLogs(ctx, evergreen.GetEnvironment().Settings().LoggerConfig.BuildloggerBaseURL, taskID, apimodels.AgentLogPrefix, LogMessageCount, t.Execution)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, err.Error())
+		}
+		agentLogs := apimodels.ReadBuildloggerToSlice(ctx, taskID, agentLogReader)
+		for i := range agentLogs {
+			agentLogPointers = append(agentLogPointers, &agentLogs[i])
+		}
+	} else {
+		taskLogs, err := model.FindMostRecentLogMessages(taskID, t.Execution, LogMessageCount, []string{},
+			[]string{apimodels.TaskLogPrefix})
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error finding task logs for task %s: %s", taskID, err.Error()))
+		}
+		//reverse
```

Comment:
FindMostRecentLogMessages already reverses it. This will put it in reverse chronological order. I believe you don't want that, correct?

Reply:
that's right. FindMostRecentLogsMessages reverses the list. But I see in the task_logg collection in production that the the order of the LogMessages are already chronological.

Reply:
This is confusing - FindTaskLogsBeforeTime does a reverse sort, so coming out of there it's reverse chronological. FindMostRecentLogsMessages reverses that so it's not chronological. Then this resolver reverses that, so it's reverse chronological again. I don't think you need to mess with it in the resolver